scatterAndGather/scatterAndGather/ScatterAndGatherViewController.swift

```diff
@@ -0,0 +1,265 @@
+//
+//  ScatterAndGatherViewController.swift
+//  scatterAndGather
+//
+//  Created by Dojo on 7/26/20.
+//  Copyright © 2020 Dojo. All rights reserved.
+//
+
+import UIKit
+
+class ScatterAndGatherViewController: UIViewController {
+    
+    
+    
+
+    @IBOutlet weak var lLabel: UILabel!
+    @IBOutlet weak var aLabel1: UILabel!
+    @IBOutlet weak var mLabel: UILabel!
+    @IBOutlet weak var bLabel: UILabel!
+    @IBOutlet weak var dLabel: UILabel!
+    @IBOutlet weak var aLabel2: UILabel!
+    
+    @IBOutlet weak var lambdaImage: UIImageView!
+    @IBOutlet weak var toggleButtonLabel: UIBarButtonItem!
+    
+    
+    
+    var arrayOfFunctions = [0, 1, 2, 3, 4, 0]
+    var allLabels: [UILabel] = []
+    var colors = [UIColor.red, UIColor.blue, UIColor.brown,
+                  UIColor.cyan, UIColor.orange, UIColor.green,
+                  UIColor.yellow]
+    private var squashPoint = CGPoint()
+    private var anticipationPoint = CGPoint()
+    
+    private var isScattered = false
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        allLabels = [lLabel,
+                     aLabel1,
+                     mLabel,
+                     bLabel,
+                     dLabel,
+                     aLabel2]
+        
+    }
+    
+    @IBAction func toggleTapped(_ sender: Any) {
+        isScattered.toggle()
+        
+        if isScattered {
+            scatterLabels()
+            toggleButtonLabel.title = "Gather"
+        } else {
+            gatherLabels()
+            toggleButtonLabel.title = "Scatter"
+        }
+    }
+    
+    
+    func scatterLabels() {
+        
+        arrayOfFunctions.shuffle()
+        var round = 0
+        
+        for letter in allLabels {
+            
+            switch arrayOfFunctions[round] {
+            case 0:
+                rotateButtonTapped(uiLabel: letter)
+            case 1:
+                springButtonTapped(uiLabel: letter)
+            case 2:
+                keyButtonTapped(uiLabel: letter)
+            case 3:
+                squashButtonTapped(uiLabel: letter)
+            default:
+                anticipationButtonTapped(uiLabel: letter)
+            }
+            round += 1
+        }
+    }
+    
+    func gatherLabels() {
+        var round = 0
+        
+        for letter in allLabels {
+            
+            switch arrayOfFunctions[round] {
+            case 0:
+                rotateButtonTapped(uiLabel: letter)
+            case 1:
+                springButtonTapped(uiLabel: letter)
+            case 2:
+                keyButtonTapped(uiLabel: letter)
+            case 3:
+                squashButtonTapped(uiLabel: letter)
+            default:
+                anticipationButtonTapped(uiLabel: letter)
+            }
+            round += 1
+        }
```

Comment:
Nice ingenuity here, Ben. That said, I would rename the functions to indicate their new use eg:

``` swift
rotate(uiLabel: letter)
```